kiwi/builder/disk.py

```diff
@@ -225,6 +186,24 @@ def create_disk(self) -> Result:
 
         :rtype: instance of :class:`Result`
         """
+        # an instance of a class with the sync_data capability
+        # representing the entire image system except for the boot/ area
+        # which could live on another part of the disk
+        system: Any = None
+
+        # an instance of a class with the sync_data capability
+        # representing the boot/ area of the disk if not part of
+        # self.system
+        system_boot: Optional[FileSystemBase] = None
+
+        # an instance of a class with the sync_data capability
+        # representing the boot/efi area of the disk
+        system_efi: Optional[FileSystemBase] = None
+
+        # an instance of a class with the sync_data capability
+        # representing the spare_part_mountpoint area of the disk
+        system_spare: Optional[FileSystemBase] = None
+
```

Comment:
They are only used within the scope of `create_disk` method, thus having them here not as object attributes is a way to ensure they are cleared once `create_disk` is done.

The counter part is having to pass them all to the utility methods as parameters.

Reply:
ok, makes sense to me, even with the enforcement to add them explicitly